site/docs/relations/physical_relations.md

```diff
@@ -71,6 +71,29 @@ The merge equijoin does a join by taking advantage of two sets that are sorted o
 | Post Join Predicate | An additional expression that can be used to reduce the output of the join operation post the equality condition. Minimizes the overhead of secondary join conditions that cannot be evaluated using the equijoin keys. | Optional, defaults true.    |
 | Join Type           | One of the join types defined in the Join operator.                                                                                                                                                                     | Required                    |
 
+
+
+## Mark Join Operator
+
+A mark join internally scans the left side, constructing a hash table that is used to mark the right side as having a join partner on the left side.  This mark can end up being True, False, or NULL.  The NULL mark is used to indicate that the right side does not have a join partner on the left side.  The mark join operator is used to implement semi-joins, anti-joins, and other join types that are not equijoins.
+
+| Signature            | Value                                                        |
+| -------------------- | ------------------------------------------------------------ |
+| Inputs               | 2                                                            |
+| Outputs              | 1                                                            |
+| Property Maintenance | Distribution is maintained. Orderedness is eliminated.       |
+| Direct Output Order  | Same as the [Join](logical_relations.md#join-operator) operator. |
```

Comment:
Does the mark join actually join two tables or does it just add the mark column?  In other words, if I apply mark join to a left table `l_key, l_payload` and a right table `r_key, r_payload` then is the output `l_key, l_payload, r_key, r_payload, mark` or is it `l_key, l_payload, mark` (which would be very strange if the left table is the build side, see previous comment)

From my read of the mark join algorithm in the paper it appears that a mark join does not actually join two tables, but only adds the mark (the algorithm scans `R` and `S`.  Only rows from `R` are inserted into the hash table `H`.  The emit at the end is a scan of `H` which only contains rows from `R`.  This is reinforced by the notation of the emit (`emit r, marker of r`)  Compare this to the full outer hash join algorithm which contains lines like `emit r,s` and `emit r,NULL`)

Reply:
You are correct - the mark join only adds the mark column, the output is `l_key, l_payload, mark` (where left is the probe side). The cardinality of the left side (probe side) is preserved - i.e. the mark join neither removes nor adds additional rows.